### PR TITLE
Fixing the branching tree file

### PR DESCRIPTION
**Pull request by artalvpes**
See the attached file as an example of the generated branching tree.
[cvrp2.pdf](https://github.com/atoptima/Coluna.jl/files/5824580/cvrp2.pdf)

### CODE REVIEW

File: src/Algorithm/treesearch.jl [hunk 1]
```diff
@@ -206,13 +206,20 @@ function print_node_in_branching_tree_file(algo::TreeSearchAlgorithm, data::Tree
     if algo.branchingtreefile !== nothing
         pb = getvalue(get_ip_primal_bound(getoptstate(data)))
         db = getvalue(get_ip_dual_bound(getoptstate(node)))
+        write(algo.branchingtreefile, read(algo.branchingtreefile, String)[1:end-1])
```
Comment (guimarqu) [inline src/Algorithm/treesearch.jl:209]: Do you want to remove the last line ? Any better way to do this ?

Comment (artalvpes) [reply src/Algorithm/treesearch.jl]: No. Only the last character (a closing brace). I pushed a change to use chop() instead.

Comment (guimarqu) [reply src/Algorithm/treesearch.jl]: Is it possible to handle the closing brace in another way than copying (almost) the whole file ?

**Comment by guimarqu** [comment]
Alternative ?

```julia
io = open("file", "r+")
seekend(io)
truncate(io, position(io)-1)  # assert that position(io) >= 1,   remove last character (last byte actually)
write!(io, "something")
close(io)
```

cf https://docs.julialang.org/en/v1/base/io-network/

**Comment by guimarqu** [comment]
> Alternative ?
> 
> ```julia
> io = open("file", "r+")
> seekend(io)
> truncate(io, position(io)-1)  # assert that position(io) >= 1,   remove last character (last byte actually)
> write!(io, "something")
> close(io)
> ```
> 
> cf https://docs.julialang.org/en/v1/base/io-network/

Even better, you don't need to truncate, you just seek for position `end - 1` and you start writing there : 
```julia
seekend(io)
n = position(io)
seek(io, n-1)
write(io, "something")
```

**Review by guimarqu — APPROVED** [review]

**Pull request merged**